src/utils/account-with-lockup.js

```diff
@@ -33,46 +33,58 @@ async function signAndSendTransaction(receiverId, actions) {
-        const poolAccountId = await this.wrappedAccount.viewFunction(lockupAccountId, 'get_staking_pool_account_id')
-        if (!poolAccountId && lockedBalance.eq(new BN(0))) {
-            console.info('Destroying lockup account to claim remaining funds', lockupAccountId)
-
-            const newKeyPair = KeyPair.fromRandom('ed25519')
-            await this.wrappedAccount.functionCall(lockupAccountId, 'add_full_access_key', {
-                new_public_key: newKeyPair.publicKey.toString()
-            }, BASE_GAS.mul(new BN(2)))
-
-            const tmpKeyStore = new InMemoryKeyStore()
-            await tmpKeyStore.setKey(this.connection.networkId, lockupAccountId, newKeyPair)
-            const tmpConnection = new Connection(this.connection.networkId, this.connection.provider, new InMemorySigner(tmpKeyStore))
-            const lockupAccount = new Account(tmpConnection, lockupAccountId)
-            await lockupAccount.deleteAccount(this.accountId)
-        } else {
-            let liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-            if (!liquidBalance.gt(missingAmount)) {
-                await this.wrappedAccount.functionCall(lockupAccountId, 'refresh_staking_pool_balance', {}, BASE_GAS.mul(new BN(3)))
-                liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-                if (!liquidBalance.gt(missingAmount)) {
-                    throw new WalletError('Not enough tokens.', 'signAndSendTransactions.notEnoughTokens')
-                }
-            }
-
-            await this.wrappedAccount.functionCall(lockupAccountId, 'transfer', {
-                // NOTE: Move all the liquid tokens to minimize transactions in the long run
-                amount: liquidBalance.toString(),
-                receiver_id: this.wrappedAccount.accountId
-            }, BASE_GAS.mul(new BN(2)))
-        }
+async function deleteLockupAccount(lockupAccountId) {
+    console.info('Destroying lockup account to claim remaining funds', lockupAccountId)
+    const newKeyPair = KeyPair.fromRandom('ed25519')
+    await this.wrappedAccount.functionCall(lockupAccountId, 'add_full_access_key', {
+        new_public_key: newKeyPair.publicKey.toString()
+    }, BASE_GAS.mul(new BN(2)))
+
+    const tmpKeyStore = new InMemoryKeyStore()
+    await tmpKeyStore.setKey(this.connection.networkId, lockupAccountId, newKeyPair)
+    const tmpConnection = new Connection(this.connection.networkId, this.connection.provider, new InMemorySigner(tmpKeyStore))
+    const lockupAccount = new Account(tmpConnection, lockupAccountId)
+    await lockupAccount.deleteAccount(this.accountId)
+}
+
+export async function transferAllFromLockup(missingAmount) {
+    let lockupAccountId = getLockupAccountId(this.accountId)
+    if (process.env.REACT_APP_USE_TESTINGLOCKUP && this.accountId.length < 64) {
```

Comment:
@Patrick1904  It makes sense to move this check inside of `getLockupAccountId` so that you get this consistent for everywhere lockup account ID is requested.

Reply:
Good point!